gcc/rust/typecheck/rust-coercion.cc

```diff
@@ -183,7 +231,12 @@ TypeCoercionRules::coerce_borrowed_pointer (TyTy::BaseType *receiver,
       return TypeCoercionRules::CoercionResult::get_error ();
     }
 
+  rust_debug ("coerce_borrowed_pointer -- autoderef cycle");
   AutoderefCycle::cycle (receiver);
+  rust_debug ("coerce_borrowed_pointer -- result: [%s] with adjustments: [%zu]",
+	      try_result.is_error () ? "failed" : "matched",
+	      try_result.adjustments.size ());
```

Comment:
I think this might cause issues on 32 bit builds
```suggestion
  rust_debug ("coerce_borrowed_pointer -- result: [%s] with adjustments: [%lu]",
	      try_result.is_error () ? "failed" : "matched",
	      (unsigned long) try_result.adjustments.size ());
```

I believe this is the fix that we had applied? Not sure

Reply:
fixed